### ISSUE TITLE
[Bug Report] master build -  rendering  color issues etc. with ryujinx

### ISSUE THREAD

**Issue by Macxine1825** [Ryujinx/Ryujinx#4325]
## Bug Report

rendering  color issues 
the problem of game save files

### What's the issue you encountered?

#rendering  color issues
 When running games in vulkan mode (or OpenGL), game rendering is likely to have abnormal color issues, the entire rendering will be overexposed and brightened.In the zelda, especially if the 'link' has an extra energy bar, If the extra energy is expended, the game will render normally.  (This seems to exist in other games, e.g. fire emblem engage,Breath of the Wild)
#the problem of game save files
The files of ryujinx are stored by two directories ' \0 ' and ' \1 ' under  '\save '  folder. The archives seem to be consistent, but the impact of reading and writing is ultimately based on the data in  ' \0 ' directory . If every time you save the data in zelda, only  data in the ' \1 ' directory will be rewritten, and  ' \0 ' will not be rewritten by mistake, which leads to the game data not being saved. (This seems to exist in other games, e.g. Persona 5 Royal Edition, kirby and the forgotten land,Breath of the Wild)

### Environment?

- Ryujinx version: 1.1.570
- vulkan mode

- Game version: 1.6.0
- the legend of Zelda Breath of the Wild

- Game version: 1.0.2
- Persona 5 Royal Edition

- System Specs:
  - OS:     win11 home consumer 21h2 22000.556
  - CPU:  Intel i7 12700h
  - GPU:  NVIDIA RTX 3060 6GB
  - RAM: 16GiB
- Applied Mods : No

### Additional context?
![IMG_20230121_035538](https://user-images.githubusercontent.com/123204719/213797691-850529f2-15af-4cd9-bd10-a5443ed94178.jpg)
![IMG_20230121_035739](https://user-images.githubusercontent.com/123204719/213797692-20b2b9a7-0810-492f-9388-80115578a6bf.jpg)

**Comment by AcK77** [comment]
Could you provide a log file please ?

**Comment by berturion** [comment]
I have this exact same issue on AMD Ryzen 5700G, AMD Radeon RX 6600, Ryujinx v1.1.574 and Zelda BOTW 1.5.0.
Green flahes, purple sometimes and image becomes completely white and never returns to normal unless I quit the game and relaunch it. But taking the same saved data, problem occurs immediately again after loading.

**Comment by yogithesymbian** [comment]
maybe this information helps on the development. i've chatted on discord but i think github is the place for this information (macOS) .
```
[mvk-info] Created 3 swapchain images with initial size (2560, 1414) and contents scale 2.0 for screen Built-in Retina Display.
00:00:32.311 N| GUI. RenderThread Gpu PrintpuInformation: Apple Apple M1 (Vulkan V1.2.231, Driver v0.2.2008) [mvk-error] VK_ERROR_INITIALIZATION_ FAILED: Render pipeline compile failed (Error code 3) :
output of type float4 is not compatible with a MTLPixelFormatR16Uint color attachment..
[mvk-info] Created 3 swapchain images with initial size (2560, 1440) and contents scale 2.0 for screen Built-in Retina Display.
```
and someone said its a missing on master build  ( allow ue4 games to work ). also i tried all games thats run on above ue4, its same like on question/issue. for example jump force deluxe edition, trek to yomi , etc.

**Comment by gdkchan** [comment]
This is not related to the bug on this issue.

**Comment by yogithesymbian** [comment]
<img width="1440" alt="color1" src="https://user-images.githubusercontent.com/28316296/216791446-d6253235-2adf-4201-b9c5-27aefc5dd528.png">
okey thanks. and this one , the picture (maybe) just a similar ( rendering color ).

**Comment by gdkchan** [comment]
> okey thanks. and this one , the picture (maybe) just a similar ( rendering color ).

It should be fixed on #4368.

**Comment by yogithesymbian** [comment]
> > okey thanks. and this one , the picture (maybe) just a similar ( rendering color ).
> 
> It should be fixed on #4368.
<img width="1440" alt="ava-ryujinx-Release-1 1 0+815009b-osx_x64" src="https://user-images.githubusercontent.com/28316296/216818625-02936ca2-89c4-4e02-8df3-76e132e2ff89.png">
thanks ,,, very good i have tried with ava-ryujinx-Release-1.1.0+815009b-osx_x64 on my device mbp m1 , and its fixed (vertex shader), should be approve to merge :D
